File: backend-api-app/src/main/java/no/war/habr/service/dto/PostDto.java
package no.war.habr.service.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;


/**
 * PostDto represents Post entity.
 *
 * @author Karachev Sasha
 */
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
public class PostDto {

    private Long id;

    private LocalDateTime created;

    private LocalDateTime updated;

    private String title;

    private String content;

    private String description;

    private String condition;

    private String owner;

    private String topic;

    private Long mainPictureId;

    private List<PictureData> pictures;

    private Set<String> tags;
}

File: frontend-app/src/app/model/post-dto.ts
import { PictureData } from "./picture-data";

export class PostDto {
    constructor(public id: number,
                public created: string,
                public updated: string,
                public title: string,
                public content: string,
                public description: string,
                public mainPictureId: number,
                public condition: string,
                public owner: string,
                public topic: string,
                public pictures: PictureData[],
                public tags: string[],
    ) {
  }
}
